tests/unit/test_repo_manager.py

```diff
@@ -34,179 +62,253 @@ def readRepo(self):
         repo.open("r")
         return repo
 
+    def init(self):
+        self.runCommand("init {}".format(self._repoPath))
+
+    def addDataset(self):
+        self._datasetName = "test_dataset"
+        cmd = "add-dataset {} {}".format(self._repoPath, self._datasetName)
+        self.runCommand(cmd)
+
+    def addReferenceSet(self):
+        self._referenceSetName = "test_rs"
+        fastaFile = paths.ncbi37FaPath
+        self.runCommand("add-referenceset {} {} --name={}".format(
+            self._repoPath, fastaFile, self._referenceSetName))
+
+    def addReadGroupSet(self):
+        bamFile = paths.bamPath
+        self._readGroupSetName = "test_rgs"
+        cmd = (
+            "add-readgroupset {} {} {} --referenceSetName={} "
+            "--name={}").format(
+            self._repoPath, self._datasetName, bamFile,
+            self._referenceSetName, self._readGroupSetName)
+        self.runCommand(cmd)
+
 
 class TestAddDataset(AbstractRepoManagerTest):
 
     def setUp(self):
         super(TestAddDataset, self).setUp()
-        self.runCommand("init {}".format(self._repoPath))
+        self.init()
 
-    def testAddDataset(self):
+    def testDefaults(self):
         name = "test_dataset"
         self.runCommand("add-dataset {} {}".format(self._repoPath, name))
         repo = self.readRepo()
         dataset = repo.getDatasetByName(name)
         self.assertEqual(dataset.getLocalId(), name)
 
-    def testAddDatasetWithSameName(self):
+    def testSameName(self):
         name = "test_dataset"
         cmd = "add-dataset {} {}".format(self._repoPath, name)
         self.runCommand(cmd)
         self.assertRaises(
+            exceptions.DuplicateNameException, self.runCommand, cmd)
+
+
+class TestAddReferenceSet(AbstractRepoManagerTest):
+
+    def setUp(self):
+        super(TestAddReferenceSet, self).setUp()
+        self.init()
+
+    def testDefaults(self):
+        fastaFile = paths.ncbi37FaPath
+        name = os.path.split(fastaFile)[1].split(".")[0]
+        self.runCommand("add-referenceset {} {}".format(
+            self._repoPath, fastaFile))
+        repo = self.readRepo()
+        referenceSet = repo.getReferenceSetByName(name)
```

Comment:
Along the same lines as another comment, it seems like we shouldn't be adding extra logic that assumes names are singular. `getByName` returns a single value, which it in fact should not. Although one could argue that references are a special case where one would never have two of the same name, names are in fact optional according to the [schemas for reference sets](https://github.com/ga4gh/schemas/blob/master/src/main/resources/avro/references.avdl#L84). We would do best to clear the way for the code to be true to the schemas, which to me would mean this function returns a list, not a single value.